service/create/service.go

```diff
@@ -1265,17 +1283,32 @@ func (s *Service) deleteFunc(obj interface{}) {
 
 	s.logger.Log("info", "deleted bucket")
 
-	// Delete policy.
-	var policy resources.NamedResource
-	policy = &awsresources.Policy{
-		ClusterID: cluster.Spec.Cluster.Cluster.ID,
-		S3Bucket:  bucketName,
-		AWSEntity: awsresources.AWSEntity{Clients: clients},
+	// Delete master policy.
+	var masterPolicy resources.NamedResource
+	masterPolicy = &awsresources.Policy{
+		ClusterID:  cluster.Spec.Cluster.Cluster.ID,
+		PolicyType: prefixMaster,
+		S3Bucket:   bucketName,
+		AWSEntity:  awsresources.AWSEntity{Clients: clients},
+	}
+	if err := masterPolicy.Delete(); err != nil {
+		s.logger.Log("error", fmt.Sprintf("%#v", err))
```

Comment:
IMO we should retry on errors and not move forward. This would also come for free with the reconciliation framework.